processing/src/main/java/org/apache/carbondata/processing/store/writer/AbstractFactDataWriter.java

```diff
@@ -418,6 +423,24 @@ protected void closeExecutorService() throws CarbonDataWriterException {
     CarbonDataWriterException exception = null;
     try {
       listener.finish();
+      String tableName = this.model.getTableName();
+      // Delete CarbonData and CarbonIndex files after copying from temp location
+      if (null != carbonDataFileTempPath && null == model.getTableSpec().getCarbonTable()
+          .getBucketingInfo(tableName)) {
+        String deleteDataAndIndexInTempPath =
```

Comment:
Why not delete this path recursively so that we do not need the traverse and delete them one by one?

You can find the delete* methods in `CarbonUtil`

Reply:
Changed the code fix. Delete carbon index and carbon data files in temp location, once it is copied from temp to target path